src/dir.c

```diff
@@ -1850,3 +1943,31 @@ read_database_map(pgBackup *backup)
 
 	return database_map;
 }
+
+/* use it to cleanup the whole PGDATA or separate tablespaces */
+//void
+//cleanup_dir(const char *path, bool follow_symlink, bool print_time)
+//{
+//	parray *files = parray_new();
+//
+//	fio_list_dir(files, path, false, follow_symlink, false, false, false, 0);
+//
+//	/* delete leaf node first */
+//	parray_qsort(files, pgFileCompareRelPathWithExternalDesc);
+//
+//	for (i = 0; i < num_files; i++)
+//	{
+//		pgFile	   *file = (pgFile *) parray_get(files, i);
+//
+//		join_path_components(full_path, backup->root_dir, file->rel_path);
+//
+//		if (interrupted)
+//			elog(ERROR, "interrupted during directory cleanup");
+//
+//		if (progress)
+//			elog(INFO, "Progress: (%zd/%zd). Delete file \"%s\"",
+//				 i + 1, num_files, full_path);
+//
+//		pgFileDelete(file->mode, full_path);
+//	}
+//}
```

Comment:
If we don't need this code anymore, let's remove it.

Reply:
Ok